components/script/script_thread.rs

```diff
@@ -1635,7 +1635,13 @@ impl ScriptThread {
     fn handle_frame_load_event(&self, parent_id: PipelineId, frame_id: FrameId, child_id: PipelineId) {
         let iframe = self.documents.borrow().find_iframe(parent_id, frame_id);
         match iframe {
-            Some(iframe) => iframe.iframe_load_event_steps(child_id),
+            Some(iframe) => {
+                // If the iframe has no "src" attribute, a load event will have already been fired
+                // when binding the iframe to the tree.
+                if iframe.upcast::<Element>().has_attribute(&local_name!("src")) {
```

Comment:
So my solution may not be the best way, the spec basically says to delay the load event if a new navigation is triggered: https://html.spec.whatwg.org/multipage/embedded-content.html#iframe-load-event-steps (See the first NOTE).

The code is not running multithreaded here, we just have asynchronous tasks. So when we queue a task to do the iframe load event steps, this will allow other things to potentially be processed first (i.e. src changing before the task is handled). We currently have a `LoadBlocker` for `Document`s that handle this. Maybe we should do something similar with `iframe`s.